src/visualstates/gui/dialogs/librariesdialog.py

```diff
@@ -81,12 +81,15 @@ def deleteButtonClicked(self):
         self.librariesChanged.emit(self.libraries)
 
     def addClicked(self):
-        self.libraries.append(self.libraryNameEdit.text())
+        input = self.libraryNameEdit.text().strip()
+        for library in self.libraries:
+            if library == input:
+                QMessageBox.warning(self, "Library Present", "Library already present in the list")
+                return
+        self.libraries.append(input)
         self.drawWindow()
         self.librariesChanged.emit(self.libraries)
```

Comment:
If there will not be any change due to duplicate entries, you should not emit any change.

Reply:
The return statement handles that